### PR TITLE
add release version guide doc

### PR DESCRIPTION
**Pull request by lepdou**
## What's the purpose of this PR

add release version guide doc

## Brief changelog

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [ ] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [ ] Update the [`CHANGES` log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by Anilople** [comment]
Also add English version?

* add file en/development/apollo-release-guide.md

* content
```
TODO
```

* update `_sidebar.md`

**Comment by nobodyiam** [comment]
Looks great! As @Anilople mentioned, we could add a placeholder document for English so that someone may translate it in the future :-)

**Review by ericzzhou — COMMENTED** [review]
good job

File: docs/zh/development/apollo-release-guide.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+
+
+填写 Release Note & 上传包
+![image.png](https://cdn.nlark.com/yuque/0/2021/png/94396/1629015996198-eb38be3c-23e2-466e-8a46-51a568cc8f30.png#clientId=u9690858a-d2d3-4&from=paste&height=930&id=ub74daa78&margin=%5Bobject%20Object%5D&name=image.png&originHeight=1860&originWidth=1518&originalType=binary&ratio=1&size=313753&status=done&style=none&taskId=u764c1d49-c82a-417f-8b5d-ca1344d8224&width=759)
+## 4.4 预发布 Apollo-Client Jar 包
+通过 github workflow 来发布。
+[https://github.com/ctripcorp/apollo/actions/workflows/release.yml](https://github.com/ctripcorp/apollo/actions/workflows/release.yml)
+![image.png](https://cdn.nlark.com/yuque/0/2021/png/94396/1629106996256-ec68519a-1b22-484e-a1bb-90376528c3a4.png#clientId=u1dee6cad-a507-4&from=paste&height=588&id=ue1658a72&margin=%5Bobject%20Object%5D&name=image.png&originHeight=1176&originWidth=2472&originalType=binary&ratio=1&size=239867&status=done&style=none&taskId=u88feb7a7-66a4-471f-ae8f-91f65eecad4&width=1236)
+注意：选择 snapshots 可能会失败，原因是账户授权问题。选择 releases 可以正常发布。
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:71]:
```suggestion
注意：如果发布的是 SNAPSHOT 版本，使用默认值 snapshots 即可，如果发布的是正式版本（不带 SNAPSHOT），则需要修改为 releases 才可以正常发布。
```

File: docs/zh/development/apollo-release-guide.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+计算 adminservice checksum
+> shasum ../apollo-adminservice/target/apollo-adminservice-1.9.0-github.zip > ../apollo-adminservice/target/apollo-adminservice-1.9.0-github.zip.sha1
+
+计算 portal checksum
+> shasum ../apollo-portal/target/apollo-portal-1.9.0-github.zip > ../apollo-portal/target/apollo-portal-1.9.0-github.zip.sha1
+
+## 4.3 创建 pre-release
+github 创建 pre-release
+![image.png](https://cdn.nlark.com/yuque/0/2021/png/94396/1629015924779-f517e179-2711-4ccb-ab46-1fb69ca2f191.png#clientId=u9690858a-d2d3-4&from=paste&height=641&id=u4161f0d9&margin=%5Bobject%20Object%5D&name=image.png&originHeight=1282&originWidth=3564&originalType=binary&ratio=1&size=258348&status=done&style=none&taskId=ue15b5715-83f8-4ff2-9c6a-285d1c11989&width=1782)
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:60]: I found sometimes the image won't display, is it better to upload the image to github?

File: docs/zh/development/apollo-release-guide.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+# 4. 版本发布
+## 4.1 打 tag
+
+1. 拉取 master 最新代码
+   1. git pull origin master  
+2. 打 tag
+   1. git tag v1.9.0
+3. push tag
+   1. git push origin tag v1.9.0
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:24]:
```suggestion
   1. git push origin tag ${new-version}
```

File: docs/zh/development/apollo-release-guide.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+则需要执行以下命令转换成 unix
+> brew install dos2unix
+> dos2unix build.sh
+
+### 4.2.3 计算构建包的 checksum
+在 scripts 目录下执行以下三个命令
+​
+
+计算 configservice checksum
+> shasum ../apollo-configservice/target/apollo-configservice-1.9.0-github.zip > ../apollo-configservice/target/apollo-configservice-1.9.0-github.zip.sha1
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:50]:
```suggestion
> shasum ../apollo-configservice/target/apollo-configservice-${new-version}-github.zip > ../apollo-configservice/target/apollo-configservice-${new-version}-github.zip.sha1
```

File: docs/zh/development/apollo-release-guide.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+计算 adminservice checksum
+> shasum ../apollo-adminservice/target/apollo-adminservice-1.9.0-github.zip > ../apollo-adminservice/target/apollo-adminservice-1.9.0-github.zip.sha1
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:53]:
```suggestion
> shasum ../apollo-adminservice/target/apollo-adminservice-${new-version}-github.zip > ../apollo-adminservice/target/apollo-adminservice-${new-version}-github.zip.sha1
```

File: docs/zh/development/apollo-release-guide.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+计算 portal checksum
+> shasum ../apollo-portal/target/apollo-portal-1.9.0-github.zip > ../apollo-portal/target/apollo-portal-1.9.0-github.zip.sha1
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:56]:
```suggestion
> shasum ../apollo-portal/target/apollo-portal-${new-version}-github.zip > ../apollo-portal/target/apollo-portal-${new-version}-github.zip.sha1
```

File: docs/zh/development/apollo-release-guide.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+## 4.5 版本发布 PMC 投票
+投票是为了让各个 PMC 成员协作验证版本的内容，防止发布有问题的版本。
+投票具体的形式为在 Discussions 发起一个帖子，可参考：[https://github.com/ctripcorp/apollo/discussions/3899](https://github.com/ctripcorp/apollo/discussions/3899)
+## 4.6 正式发布 Apollo-Client Jar 到仓库
+
+
+## 4.7 发布 Docker 镜像
+### 4.7.1 构建镜像
+在 4.2 步骤打完包的前提下，在 apollo 根目录下执行
+> mvn docker:build -pl apollo-configservice,apollo-adminservice,apollo-portal -D1.9.0
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:81]:
```suggestion
> mvn docker:build -pl apollo-configservice,apollo-adminservice,apollo-portal -D${new-version}
```

File: docs/zh/development/apollo-release-guide.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+如果报以下错误：
+> zsh: ./build.sh: bad interpreter: /bin/sh^M: no such file or directory
+
+则需要执行以下命令转换成 unix
+> brew install dos2unix
+> dos2unix build.sh
+
+### 4.2.3 计算构建包的 checksum
+在 scripts 目录下执行以下三个命令
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:46]: 貌似在 scripts 目录下执行的话，sha1 文件中的内容也会带上很长的相对路径，如

d6248f187d9e62bbde210061911836bf815d4f8a  ../apollo-configservice/target/apollo-configservice-1.9.1-github.zip

Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md:46]: done

File: docs/zh/development/apollo-release-guide.md (hunk 7 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

File: docs/zh/development/apollo-release-guide.md (hunk 6 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

File: docs/zh/development/apollo-release-guide.md (hunk 5 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

File: docs/zh/development/apollo-release-guide.md (hunk 4 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

File: docs/zh/development/apollo-release-guide.md (hunk 3 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

File: docs/zh/development/apollo-release-guide.md (hunk 2 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

File: docs/zh/development/apollo-release-guide.md (hunk 1 above)
Comment (lepdou) [reply docs/zh/development/apollo-release-guide.md]: done

**Comment by lepdou** [comment]
I have added english doc placeholder.

File: docs/zh/development/apollo-release-guide.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:46]: This line is not necessary.

File: docs/zh/development/apollo-release-guide.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+> zsh: ./build.sh: bad interpreter: /bin/sh^M: no such file or directory
+
+则需要执行以下命令转换成 unix
+> brew install dos2unix
+> dos2unix build.sh
+
+### 4.2.3 计算构建包的 checksum
+
+计算 configservice checksum
+>在 ${apolloWorkspaceRootPath}/apollo-configservice/target/ 目录下执行：
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:48]:
```suggestion
>在 ${apollo_home}/apollo-configservice/target/ 目录下执行：
```

File: docs/zh/development/apollo-release-guide.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+> 
+>shasum apollo-configservice-${new-version}-github.zip > apollo-configservice-${new-version}-github.zip.sha1
+
+计算 adminservice checksum
+>在 ${apolloWorkspaceRootPath}/apollo-adminservice/target/ 目录下执行：
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:53]:
```suggestion
>在 ${apollo_home}/apollo-adminservice/target/ 目录下执行：
```

File: docs/zh/development/apollo-release-guide.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+>
+>shasum apollo-adminservice-${new-version}-github.zip > apollo-adminservice-${new-version}-github.zip.sha1
+
+计算 portal checksum
+>在 ${apolloWorkspaceRootPath}/apollo-portal/target/ 目录下执行：
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:58]:
```suggestion
>在 ${apollo_home}/apollo-portal/target/ 目录下执行：  
```

File: docs/zh/development/apollo-release-guide.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+>
+> shasum apollo-portal-${new-version}-github.zip > apollo-portal-${new-version}-github.zip.sha1
+
+## 4.3 创建 pre-release
+github 创建 pre-release
+![image.png](https://raw.githubusercontent.com/ctripcorp/apollo/master/doc/images/local-development/create-release.png)
+
+
+
+
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:68]:
```suggestion
```

File: docs/zh/development/apollo-release-guide.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+填写 Release Note & 上传包
+![image.png](https://raw.githubusercontent.com/ctripcorp/apollo/master/doc/images/local-development/fill-release-form.png)
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:70]:
```suggestion
![image.png](https://raw.githubusercontent.com/ctripcorp/apollo/master/doc/images/local-development/fill-release-form.png)

```

File: docs/zh/development/apollo-release-guide.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+## 4.4 预发布 Apollo-Client Jar 包
+通过 github workflow 来发布。
+[https://github.com/ctripcorp/apollo/actions/workflows/release.yml](https://github.com/ctripcorp/apollo/actions/workflows/release.yml)
+![image.png](https://raw.githubusercontent.com/ctripcorp/apollo/master/doc/images/local-development/publish-sdk.png)
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:74]:
```suggestion

![image.png](https://raw.githubusercontent.com/ctripcorp/apollo/master/doc/images/local-development/publish-sdk.png)

```

File: docs/zh/development/apollo-release-guide.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+> 注意：如果发布的是 SNAPSHOT 版本，使用默认值 snapshots 即可，如果发布的是正式版本（不带 SNAPSHOT），则需要修改为 releases 才可以正常发布。
+## 4.5 版本发布 PMC 投票
```
Comment (nobodyiam) [inline docs/zh/development/apollo-release-guide.md:76]:
```suggestion
## 4.5 版本发布 PMC 投票

```

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**